ui/app/components/secure-variable-form.js

```diff
@@ -94,19 +85,29 @@ export default class SecureVariableFormComponent extends Component {
     ]);
   }
 
-  @action
-  validatePath(e) {
-    const value = trimPath([e.target.value]);
+  /**
+   * @typedef {Object} DuplicatePathWarning
+   * @property {string} path
+   */
+
+  /**
+   * @type {DuplicatePathWarning}
+   */
+  @computed('args.{model.path,existingVariables}', 'variableNamespace')
+  get duplicatePathWarning() {
     const existingVariables = this.args.existingVariables || [];
+    const pathValue = trimPath([this.args.model.path]);
     let existingVariable = existingVariables
       .without(this.args.model)
-      .find((v) => v.path === value);
+      .find(
+        (v) => v.path === pathValue && v.namespace === this.variableNamespace
+      );
```

Comment:
`suggestion`: the `filter` method takes a callback, too, if you want to play code golf.